genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/InitialSetupTask.java

```diff
@@ -329,4 +358,22 @@ private void createJobEnvironmentVariables(
         // Append new line
         writer.write(LINE_SEPARATOR);
     }
+
+    private String tagsToString(final Set<String> tags) {
```

Comment:
Get rid of this whole method.

Use [this](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#join-java.lang.Iterable-char-)

Also should we try to bash escape any existing commas?